lib/mongoid/criteria/queryable/extensions/date.rb

```diff
@@ -23,9 +23,9 @@ def __evolve_date__
           # @example Evolve the date.
           #   date.__evolve_time__
           #
-          # @return [ Time ] The date as a local time.
+          # @return [ Time | ActiveSupport::TimeWithZone ] The date as a local time.
           def __evolve_time__
-            ::Time.local(year, month, day)
+            ::Time.configured.local(year, month, day)
```

Comment:
This code is now consistent with the `__mongoize_time__` behavior. There is no issue making this return `ActiveSupport::TimeWithZone` as the Ruby driver has no problem with it. However, we could suffix this with `.to_time` to avoid changing return type, but I think it's better to just be consistent with Mongoize.

Reply:
AFAIK this code is correct as-is. Note `Time.configured` is a Mongoid-defined method which handles the branching for `use_activesupport_time_zone`.

It is not necessary to call `use_utc` here, as it only affects how time values are demongoized. From the Mongoid docs:

> If true, times stored in MongoDB will be returned in UTC.... The use_utc setting does not affect how times are parsed - parsing is always done in local time when the input being parsed does not include time zone information.